src/proxy/connection_manager.rs

```diff
@@ -118,6 +133,19 @@ impl ConnectionManager {
         // potentially large copy under read lock, could require optomization
         self.drains.read().await.keys().cloned().collect()
     }
+
+    /// fetch looks up a tuple and returns the connection metadata
+    // TODO: we should key on tuple so we can more efficiently lookup
```

Comment:
Maybe another index instead of re-keying? I suppose we could see about keying on tuple and storing the full `ProxyRbacContext` in the `ConnectionDrain` struct. Otherwise we'll need to do work to re-construct the full context when we need to assert.

Reply:
Yeah I would probably do the index if we do it